src/main/java/com/influxdb/v3/client/internal/VectorSchemaRootConverter.java

```diff
@@ -95,45 +97,141 @@ PointValues toPointValues(final int rowNumber,
             String valueType = parts[2];
 
             if ("field".equals(valueType)) {
-                p.setField(name, value);
+                setFieldWithMetaType(p, name, value, metaType);
             } else if ("tag".equals(valueType) && value instanceof String) {
                 p.setTag(name, (String) value);
             } else if ("timestamp".equals(valueType)) {
-                setTimestamp(value, schema, p);
+                var timeNano = NanosecondConverter.getTimestampNano(value, field);
+                p.setTimestamp(timeNano, WritePrecision.NS);
             }
         }
         return p;
     }
 
-    private void setTimestamp(@Nonnull final Object value,
-                              @Nonnull final Field schema,
-                              @Nonnull final PointValues pointValues) {
-        if (value instanceof Long) {
-            if (schema.getFieldType().getType() instanceof ArrowType.Timestamp) {
-                ArrowType.Timestamp type = (ArrowType.Timestamp) schema.getFieldType().getType();
-                TimeUnit timeUnit;
-                switch (type.getUnit()) {
-                    case SECOND:
-                        timeUnit = TimeUnit.SECONDS;
+    /**
+     * Set field value for PointValues base on iox::column::type.
+     *
+     * @param p    The target PointValues.
+     * @param fieldName       Field name in PointValues
+     * @param value The value to be set
+     * @param metaType The iox::column::type column meta type,
+     *                 eg: iox::column_type::field::integer, iox::column_type::field::float
+     */
+    public void setFieldWithMetaType(final PointValues p,
+                                      final String fieldName,
+                                      final Object value,
+                                      final String metaType) {
+        if (value == null) {
+            return;
+        }
+
+        switch (metaType) {
+            case "iox::column_type::field::integer":
+            case "iox::column_type::field::uinteger":
+                if (value instanceof Long) {
+                    p.setIntegerField(fieldName, TypeCasting.toLongValue(value));
+                } else {
+                    p.setNullField(fieldName);
+                    LOG.warning(String.format("Value of %s is not an Integer", fieldName));
+                }
+                break;
+            case "iox::column_type::field::float":
+                if (value instanceof Double) {
```

Comment:
toDoubleValue is able to handle Number:
```suggestion
                if (value instanceof Number) {
```